sqlmesh/core/engine_adapter/base.py

```diff
@@ -1405,6 +1406,29 @@ def rename_table(
                 )
         self.execute(exp.rename_table(old_table_name, new_table_name))
 
+    def list_data_objects(
```

Comment:
should we make this consistent with the private name? Seems strange to have `list` and then `get`.